financial_calculations/coupon_rates.py

```diff
@@ -60,6 +58,6 @@ def calculate_coupon_rate(start_date, maturity_years, par_value, forward_curve):
         final_discount = discount_factors[-1]   
 
         # Calculate the coupon rate
-        coupon_rate = par_value * (initial_discount - final_discount) / (annuity * par_value)
+        coupon_rate = (initial_discount - final_discount) / annuity
```

Comment:
turns out you don't need par value!  the more general approach is to have a "face" amount and a market value.  right now we assume that face == market value, so it cancels out.  however bond prices change through time, but the face amount doesn't.  at least for a bullet